internal/envoy/route.go

```diff
@@ -269,12 +279,12 @@ func weightedClusters(clusters []*dag.Cluster) *envoy_api_v2_route.WeightedClust
 		}
 		wc.Clusters = append(wc.Clusters, c)
 	}
-	// Check if no weights were defined, if not default to even distribution
+	// Check if no weights were defined, default to even distribution
 	if total == 0 {
 		for _, c := range wc.Clusters {
 			c.Weight.Value = 1
 		}
-		total = uint32(len(clusters))
+		total = uint32(len(wc.Clusters))
```

Comment:
So at this point, the mirrored Cluster is part of the slice. I don't think that it should be included in the slice to calculate weights.

Reply:
It's not part of the slice, any mirrored cluster never got added to `wc.Clusters` so not taken into the weight calculation.